pkg/kubelet/network/cni/cni.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cni
+
+import (
+	"fmt"
+	"github.com/appc/cni"
+	cniTypes "github.com/appc/cni/pkg/types"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/kubelet/network"
+	kubeletTypes "k8s.io/kubernetes/pkg/kubelet/types"
+	"net"
+	"sort"
+)
+
+const (
+	DefaultPluginName    = "kubernetes-bridge"
+	DefaultNetDir        = "/etc/cni/net.d"
+	DefaultCNIDir        = "/opt/cni/bin"
+	VendorCNIDirTemplate = "/opt/%s/bin"
+)
+
+type cniNetworkPlugin struct {
+	name          string
+	NetworkConfig *cni.NetworkConfig
+	CNIConfig     *cni.CNIConfig
+	host          network.Host
+}
+
+func ProbeNetworkPlugins(pluginDir string) []network.NetworkPlugin {
+
+	configList := make([]network.NetworkPlugin, 0)
+
+	files, err := cni.ConfFiles(pluginDir)
+	switch {
+	case err != nil:
+		return configList
+	case files == nil || len(files) == 0:
+		return configList
+	}
+
+	sort.Strings(files)
+	for _, confFile := range files {
+		conf, err := cni.ConfFromFile(confFile)
```

Comment:
From offline discussion - which model do we really want for this?  Does the --network-plugin flag mean:

1) `--network-plugin="cni"` => use whatever CNI config(s) are present on the machine, make that a different problem
2) `--network-plugin="foobar"` => use only the 'foobar' CNI config
3) `--network-plugin="bridge"` => use the CNI 'bridge' driver (and make up a config?)

I think that, for now, (1) is simpler and more useful.  In the fullness of time we may want to define Network objects in our API and use those to define CNI configs, and then allow Pods to specify which Networks they join - but I don't think we should start there.

So I think (argue with me) this registration should be a single plugin called "cni".  Over time we can EOL kubelet's concept of driver plugins entirely and devolve into a question of "should kubelet set up networking itself or should it defer to the runtime"?  And in fact, it's even more specific - "should kubelet's docker runtime set up networking itself or should it defer to docker".  I imagine we'll always want rkt to do it's own network setup, since it has totally different hooks than docker anyway.